pkg/awsvpc/parser.go

```diff
@@ -0,0 +1,344 @@
+		return nil, fmt.Errorf("error merging input vpc resources files")
+	}
+	rc1.VpcsList = append(rc1.VpcsList, rc2.VpcsList...)
+	rc1.SubnetsList = append(rc1.SubnetsList, rc2.SubnetsList...)
+	rc1.NetworkACLsList = append(rc1.NetworkACLsList, rc2.NetworkACLsList...)
+	rc1.SecurityGroupsList = append(rc1.SecurityGroupsList, rc2.SecurityGroupsList...)
+	rc1.InternetGWList = append(rc1.InternetGWList, rc2.InternetGWList...)
+	rc1.InstancesList = append(rc1.InstancesList, rc2.InstancesList...)
+
+	return rc1, nil
+}
+
+func (rc *AWSresourcesContainer) VpcConfigsFromFiles(fileNames []string, vpcID, resourceGroup string, regions []string) (
+	*vpcmodel.MultipleVPCConfigs, error) {
+	for _, file := range fileNames {
+		mergedRC := &AWSresourcesContainer{}
+		err1 := mergedRC.ParseResourcesFromFile(file)
+		if err1 != nil {
+			return nil, fmt.Errorf("error parsing input vpc resources file: %w", err1)
+		}
+		rc, err1 = mergeResourcesContainers(mergedRC, rc)
+		if err1 != nil {
+			return nil, err1
+		}
+	}
+	vpcConfigs, err2 := rc.VPCConfigsFromResources(vpcID, resourceGroup, regions)
+	if err2 != nil {
+		return nil, fmt.Errorf("error generating cloud config from input vpc resources file: %w", err2)
+	}
+	return vpcConfigs, nil
+}
+
+func (rc *AWSresourcesContainer) filterByVpc(vpcID string) map[string]bool {
+	shouldSkipVpcIds := make(map[string]bool)
+	for _, vpc := range rc.VpcsList {
+		if vpcID != "" && *vpc.VpcId != vpcID {
+			shouldSkipVpcIds[*vpc.VpcId] = true
+		}
+	}
+	return shouldSkipVpcIds
+}
+
+// VPCConfigsFromResources returns a map from VPC UID (string) to its corresponding VPCConfig object,
+// containing the parsed resources in the relevant model objects
+func (rc *AWSresourcesContainer) VPCConfigsFromResources(vpcID, resourceGroup string, regions []string) (
+	*vpcmodel.MultipleVPCConfigs, error) {
+	res := vpcmodel.NewMultipleVPCConfigs("AWS Cloud") // map from VPC UID to its config
+	var err error
+
+	// map to filter resources, if certain VPC, resource-group or region list to analyze is specified,
+	// skip resources configured outside that VPC
+	shouldSkipVpcIds := rc.filterByVpc(vpcID)
+
+	err = rc.getVPCconfig(res, shouldSkipVpcIds)
+	if err != nil {
+		return nil, err
+	}
+
+	var vpcInternalAddressRange map[string]*ipblock.IPBlock // map from vpc name to its internal address range
+
+	subnetNameToNetIntf := map[string][]*commonvpc.NetworkInterface{}
+	netIntfToSGs := map[string][]types.GroupIdentifier{}
+	err = rc.getInstancesConfig(subnetNameToNetIntf, netIntfToSGs, res, shouldSkipVpcIds)
+	if err != nil {
+		return nil, err
+	}
+	vpcInternalAddressRange, err = rc.getSubnetsConfig(res, subnetNameToNetIntf, shouldSkipVpcIds)
+	if err != nil {
+		return nil, err
+	}
+	// assign to each vpc object its internal address range, as inferred from its subnets
+	err = commonvpc.UpdateVPCSAddressRanges(vpcInternalAddressRange, res)
+	if err != nil {
+		return nil, err
+	}
+
+	err = rc.getSGconfig(res, shouldSkipVpcIds, netIntfToSGs)
```

Comment:
why aren't the nacl configs handled here with the sg's?  where are they handled?

Reply:
not supported yet, follow-up pr